client/my-sites/plugins/plugins-discovery-page/index.jsx

```diff
@@ -146,38 +171,17 @@ const PluginsDiscoveryPage = ( props ) => {
 		category: 'featured',
 	} );
 
-	const requiredPlansPurchasedPlugins = useSelector( ( state ) =>
-		getPlansForFeature( state, props.selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
-	);
-	const requiredPlansAllPlugins = useSelector( ( state ) =>
-		getPlansForFeature( state, props.selectedSite?.ID, FEATURE_INSTALL_PLUGINS )
-	);
-
-	const hasInstallPurchasedPlugins = useSelector( ( state ) =>
-		siteHasFeature( state, props.selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
-	);
-
-	// Whether to show an upgrade banner specific to premium plugins.
-	const lowerPlanAvailable =
-		! hasInstallPurchasedPlugins &&
-		requiredPlansPurchasedPlugins[ 0 ] !== requiredPlansAllPlugins[ 0 ];
-
 	return (
 		<>
 			{ ! props.jetpackNonAtomic && (
 				<>
 					<div className="plugins-discovery-page__upgrade-banner">
-						{ ! hasInstallPurchasedPlugins && lowerPlanAvailable && (
-							<UpgradeNudgePaid { ...props } />
-						) }
-						{ ! hasInstallPurchasedPlugins && ! lowerPlanAvailable && (
-							<UpgradeNudge { ...props } />
-						) }
+						<UpgradeNudge { ...props } paidPlugins={ true } />
 					</div>
 					<PaidPluginsSection { ...props } />
 				</>
 			) }
-			{ ( hasInstallPurchasedPlugins || lowerPlanAvailable ) && <UpgradeNudge { ...props } /> }
+			{ <UpgradeNudge { ...props } /> }
```

Comment:
Can we make this component so it looks like:
```
...
<UpgradeNudge />
<PaidPluginSection />
<FeaturedPluginSection />
...
```

I'm thinking it'll be easier to follow if we move all the `isJetpack ? null : <Component>` display logic into the components themselves, especially in the nudge case where we're doing further should I null checks.

Reply:
✅